tools/autograd/gen_python_functions.py

```diff
@@ -165,6 +168,20 @@
     .pinned_memory(${pin_memory});
 """)
 
+INPLACE_DEPRECATION_MESSAGE = ('TORCH_WARN("In-place functions such as torch.{name} are deprecated '
+                               'and will be removed in the next release. '
+                               'Please use their in-place method counterparts (Tensor.{name}) instead.");')
+
+# These functions should not exist because they are in-place. We now deprecate them before removing them.
+DEPRECATED_INPLACE_FUNCTIONS = ('abs_', 'acos_', 'addmv_', 'as_strided_', 'asin_', 'atan_', 'ceil_', 'celu_',
+                                'clamp_', 'clamp_max_', 'clamp_min_', 'cos_', 'cosh_', 'detach_',
+                                'embedding_renorm_', 'erf_', 'erfc_', 'exp_', 'expm1_', 'fill_', 'floor_', 'frac_',
+                                'index_put_', 'log10_', 'log1p_', 'log2_', 'log_', 'neg_', 'reciprocal_', 'relu_',
```

Comment:
how did you determine this list?  For example, it looks like rrelu_ is used in nn/functional, but it's not in the EXEMPTED list.

Did you check that none of these methods are actually used?

Reply:
I'm still unsure yet what to do with `relu`, `rrelu`, and `celu`. These look like nn functions but they are under torch, and their positioning is kind of vague. For all others, I add them to deprecation list because they are in-place and semantically shouldn't be in nn.